lib/Semantics/resolve-names.cpp

```diff
@@ -5094,6 +5119,25 @@ void ConstructVisitor::Post(const parser::TypeGuardStmt::Guard &x) {
   }
 }
 
+void ConstructVisitor::Post(const parser::SelectRankCaseStmt::Rank &x) {
+  if (auto *symbol{MakeAssocEntity()}) {
+    SetTypeFromAssociation(*symbol);
+    SetAttrsFromAssociation(*symbol);
+    std::visit(
```

Comment:
When testing for only one alternative, do not use `visit()`.  `std::get_if<T>(&x.u)` is much better.